tests/eth2/fixtures/state-fixtures/test_genesis_initialization.py

```
@@ -0,0 +1,118 @@
+    Tuple,
+)
+from dataclasses import (
+    dataclass,
+    field,
+)
+import pytest
+
+from eth_utils import (
+    decode_hex,
+)
+from eth_typing import (
+    Hash32,
+)
+from ssz.tools import (
+    from_formatted_dict,
+)
+
+from eth2.beacon.genesis import (
+    initialize_beacon_state_from_eth1,
+)
+from eth2.beacon.tools.misc.ssz_vector import (
+    override_lengths,
+)
+from eth2.beacon.types.deposits import Deposit
+from eth2.beacon.types.states import BeaconState
+from eth2.beacon.tools.fixtures.config_name import (
+    ONLY_MINIMAL,
+)
+from eth2.beacon.tools.fixtures.helpers import (
+    validate_state,
+)
+from eth2.beacon.tools.fixtures.loading import (
+    get_bls_setting,
+    get_deposits,
+)
+from eth2.beacon.tools.fixtures.test_case import (
+    BaseTestCase,
+)
+from eth2.beacon.typing import (
+    Timestamp,
+)
+
+from tests.eth2.fixtures.helpers import (
+    get_test_cases,
+)
+from tests.eth2.fixtures.path import (
+    BASE_FIXTURE_PATH,
+    ROOT_PROJECT_DIR,
+)
+
+
+# Test files
+RUNNER_FIXTURE_PATH = BASE_FIXTURE_PATH / 'genesis'
+HANDLER_FIXTURE_PATHES = (
+    RUNNER_FIXTURE_PATH / 'initialization',
+)
+FILTERED_CONFIG_NAMES = ONLY_MINIMAL
+
+
+#
+#  Test format
+#
+@dataclass
+class GenesisInitializationTestCase(BaseTestCase):
+    bls_setting: bool
+    description: str
+    eth1_block_hash: Hash32
+    eth1_timestamp: Timestamp
+    state: BeaconState
+    deposits: Tuple[Deposit, ...] = field(default_factory=tuple)
+
+
+#
+# Helpers for generating test suite
+#
+def parse_genesis_initialization_test_case(test_case, config):
+    override_lengths(config)
+
+    bls_setting = get_bls_setting(test_case)
+    eth1_block_hash = decode_hex(test_case['eth1_block_hash'])
+    eth1_timestamp = test_case['eth1_timestamp']
+    state = from_formatted_dict(test_case['state'], BeaconState)
+    deposits = get_deposits(test_case, Deposit)
+
+    return GenesisInitializationTestCase(
+        line_number=test_case.lc.line,
```

Wait ..., if we are using `YAML(typ='unsafe')`, `test_case.lc.line` should raise an exception